docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1739,212 @@ As a response to blocked request AdGuard returns a short video placeholder.
+
+#### 2. Defined content-type modifiers, defined methods, $popup, special exceptions: {#priority-category-2}
+
+All allowed content types:
+[//]: # (Please keep them sorted)
+
+* [`$document`](#document-modifier),
+* [`$font`](#font-modifier),
+* [`$image`](#image-modifier),
+* [`$media`](#media-modifier),
+* [`$object`](#object-modifier),
+* [`$other`](#other-modifier),
+* [`$ping`](#ping-modifier),
+* [`$script`](#script-modifier),
+* [`$stylesheet`](#stylesheet-modifier),
+* [`$subdocument`](#subdocument-modifier),
+* [`$websocket`](#websocket-modifier),
+* [`$xmlhttprequest`](#xmlhttprequest-modifier);
+
+This also includes rules that implicitly add the modifier `$document`:
+* [`$popup`](#popup-modifier);
+
+or special exceptions that implicitly add `$document,subdocument`:
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier)';
+
+Or allowed methods via [`$method`](#method-modifier).
+
+The presence of any content-type modifiers adds `(50 + 50 / N)`, where `N` is the number of modifiers present, for example:
+`||example.com^$image,script` will add `50 + 50 / 2 = 50 + 25 = 75` to the total weight of the rule. The `$popup` also belongs to this category, because it implicitly adds the modifier `$document`. Similarly, specific exceptions add `$document,subdocument`.
+
+If there is a `$method` modifier in the rule with allowed methods it adds `(50 + 50 / N)`, where `N` is the number of methods allowed, for example:
+`||example.com^$method=GET|POST|PUT` will add `50 + 50 / 3 = 50 + 16.6 = 67` to the total weight of the rule.
+
+#### 3. $domain or $app with allowed domains or applications: {#priority-category-3}
+
+Specified domains through `$domain` or specified applications through `$app` add `100 + 100 / N`, where `N` is the number of modifier values for example:
+`||example.com^$domain=example.com|example.org|example.net` will add `100 + 100 / 3 = 134.3 = 135` or
+`||example.com^$app=org.example.app1|org.example.app2` will add `100 + 100 / 2 = 151` or
+`||example.com^$domain=example.com,app=org.example.app1|org.example.app2` will add `100 + 100/1` ($domain part) and `100 + 100/2` ($app part) - will add `350` in total.
+
+Modifier values that are regexps or tld will be interpreted as normal entries of the form `example.com` and counted one by one, for example:
+`||example.com^$domain=example.*` will add `100 + 100 / 1 = 200` or
+`||example.com^$domain=example.*|adguard.*` will add `100 + 100 / 2 = 150`.
+
+#### 4. $redirect rules {#priority-category-6}
```

Comment:
Whilte implementing the new spec, I discovered that in CoreLibs we support a syntax (which is seemingly never mentioned in this knowledge base) for specifying a priority for redirect rules, e.g.: `$redirect=noopjs:42` is higher priority than `$redirect=noopjs:-3`. It is currently used only when comparing redirect rules to each other, and before comparing the "common properties" of the redirect rules. Are we keeping this feature? If so, how should this redirect-rule specific priority be integrated into the new priority framework?

Reply:
@ngorskikh Yes, we discussed them in a conversation with you two weeks ago. As far as I remember, they will only be added for compatibility with uBlock syntax. As for the extension - we don't support those specified weights and, as far as I know, we don't intend to.
As for CoreLibs - it might be more correct to compare $redirect rules according to our weights priorities first and only after that, if there are two rules with the same weight - CoreLibs can use these extra weights as part of "[specific implementation logic](https://github.com/AdguardTeam/KnowledgeBase/pull/196/files/0cd37507c8ef08d53bcc868141c2c2e7a4e1971f#diff-190ba525f7c4d0b0eda33dab9df98df01d3ac072d6aeef951f044c96d58a25c9R1746)".

Reply:
Actually, it is [mentioned](https://github.com/AdguardTeam/KnowledgeBase/blob/master/docs/general/ad-filtering/create-own-filters.md?plain=1#L1374)

Reply:
Actually, it was mentioned in the [old KB](https://github.com/AdguardTeam/AdguardKnowledgeBase/blob/master/01.general/02.how-to-create-your-own-ad-filters/docs.en.md?plain=1#L1026) but not here 🥲

Reply:
We discussed this and came to a conclusion that this is a really rarely required thing and the only reason for it is that uBO does not have the concept of rule priority.

So we decided to not support it in tsurlfilter and simply "strip" the priority part when doing the conversion.
Please do the same in CoreLibs.

@105th I think we should mention it in the documentation:

> `$redirect in uBlock Origin supports specifying priority, e.g. `$redirect=noopjs:42`. AdGuard does not support it and instead just discards the priority postfix.`